src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTab.java

```diff
@@ -274,6 +290,32 @@ public void onFailure(Throwable caught) {
 		}
 	}
 	
+	 public void configureStorageLocation(String containerEntityId) {
+		 synapseClient.getUploadDestinations(containerEntityId, new AsyncCallback<List<UploadDestination>>() {
```

Comment:
this looks a lot like the code up in the EntityPageTop.  should this instead be a method in EntityMetadata (where you can request it to get and show the storage location).